python/setup.py

```diff
@@ -495,32 +491,10 @@ def has_ext_modules(foo):
     use_scm_version={
         'root': os.path.dirname(setup_dir),
         'parse': parse_git,
-        'write_to': os.path.join(scm_version_write_to_prefix,
+        'version_file': os.path.join(scm_version_write_to_prefix,
                                  'pyarrow/_generated_version.py'),
```

Comment:
Looking at the looks this isn't allowed to be an absolute path. Maybe:
```suggestion
        'version_file': 'pyarrow/_generated_version.py',
```

Reply:
Yeah, locally that works. I didn't setup with a proper arrow install so cmake fails but should work in ci :)